Automattic/wp-calypso
# Stats: Flesh out highlight card components

Related to #93973. I apologize for the large LOC; ~half of it is from Storybook additions/edits.

## Proposed Changes

Before|After
-|-
![image](https://github.com/user-attachments/assets/ccfe07e7-11c6-4f27-b13b-20d06fc93fb4)|![image](https://github.com/user-attachments/assets/0a613916-a3bb-4965-9ee8-733c4f639851)
![image](https://github.com/user-attachments/assets/36264486-bdac-4b2b-b269-3605d17da96c)|![image](https://github.com/user-attachments/assets/784350a3-8e51-4990-b05e-a2b81e4380fd)

- Adds tooltip functionality to `<CountCard>`.
- Replaced usage of `<CountComparisonCard>` with `<CountCard>` where no comparisons were taking place.
- Refactored number formatting manipulation into a dedicated library module.
- Standardized percentage formatting using `Intl.NumberFormat`.
- Updated stories for each highlight card component, which unfortunately bloated the LOC for this PR.

## Why are these changes being made?

These changes make it more obvious when CountCards and CountComparisonCards should be used. If we want to showcase a single value, CountCards should be used; if we want to visualize how a value has changed over time, it should be CountComparisonCards.

These changes also standardize how we perform international number formatting in highlight card components.

## Testing Instructions


* Open the live branch for this change and ensure that the highlight cards render and behave as expected on the following pages for **desktop and mobile viewports**:
    * Traffic dashboard (`/stats/day/example.com`).
    * Insights dashboard (`/stats/insights/example.com`).
    * Subscribers dashboard (`/stats/subscribers/example.com`).
* Open Storybook and ensure that each highlight card component is easily explorable.

## Pre-merge Checklist
